src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker.cs

```diff
@@ -174,11 +174,15 @@ SymbolKind.NetModule or
                     {
                         // We may still have a cached generator; we'll have to remember to run generators again since we are making some
                         // change here. We'll also need to update the other state of the driver if appropriate.
-                        var generatorInfo = state.GeneratorInfo.WithDocumentsAreFinal(false);
```

Comment:
this line seems like it would do nothing.  i've written below whu i believe that.  and i've now moved to having contracts here.  this is all in service of making it very clear how this bit transitions through the state machine we have here.